### PR TITLE
add: ctwm

### LINKED ISSUE #1030: [Request] CTWM
**What software are you requesting to be packaged?**
CTWM https://www.ctwm.org/index.html

**Describe the motivation**

Funny window manager for this set of dotfiles
http://scarlettppcblog.ddns.net/guides/ctwmpinkv1.html

**Additional context**
It's in RPMSphere already?

### PR DESCRIPTION
**Pull request by nothingneko**
Closes #1030

### CODE REVIEW

File: anda/desktops/ctwm/ctwm.spec [hunk 1]
```diff
@@ -0,0 +1,54 @@
+Summary: Lightweight window manager with virtual workspaces
+Name: ctwm
+Version: 4.1.0
+Release: 1
+URL: https://ctwm.org
+BuildRequires: libx11-devel libjpg-devel libxt-devel libxext-devel libxmu-devel libxpm-devel cmake
+Source0: https://www.ctwm.org/dist/%{name}-%{version}.tar.gz
+Source1: %{name}.desktop
```
Comment (madonuko) [inline anda/desktops/ctwm/ctwm.spec:8]: Doesn't exist in the same directory as the spec

File: anda/desktops/ctwm/ctwm.spec [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+License: MIT
+Requires: m4
+# Derived from RPMSphere's packaging
+
+%description
+CTWM is a window manager based on TWM (with virtual workspaces added)
```
Comment (madonuko) [inline anda/desktops/ctwm/ctwm.spec]: full stop maybe

File: anda/desktops/ctwm/ctwm.spec [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+%cmake
+%cmake_build
+
+%install
+%cmake_install
+%{__install} -Dm644 %{SOURCE1} $RPM_BUILD_ROOT%{_datadir}/xsessions/%{name}.desktop
+
+%clean
+rm -rf $RPM_BUILD_ROOT
+
```
Comment (madonuko) [inline anda/desktops/ctwm/ctwm.spec]:
```suggestion
```
You don't need this

File: anda/desktops/ctwm/ctwm.spec [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+%files
+%doc README.md CHANGES.md COPYRIGHT
```
Comment (madonuko) [inline anda/desktops/ctwm/ctwm.spec]:
```suggestion
%doc README.md CHANGES.md
%license COPYRIGHT
```

File: anda/desktops/ctwm/ctwm.spec (hunk 1 above)
Comment (madonuko) [reply anda/desktops/ctwm/ctwm.spec:8]: you probably forgot to copy this from RPMSphere

File: anda/desktops/ctwm/ctwm.spec (hunk 4 above)
Comment (madonuko) [reply anda/desktops/ctwm/ctwm.spec]: also I looked at upstream and they somehow use .adoc instead of .md

File: anda/desktops/ctwm/ctwm.spec [hunk 5]
```diff
@@ -0,0 +1,54 @@
+Summary: Lightweight window manager with virtual workspaces
+Name: ctwm
+Version: 4.1.0
+Release: 1
+URL: https://ctwm.org
+BuildRequires: libx11-devel libjpg-devel libxt-devel libxext-devel libxmu-devel libxpm-devel cmake
+Source0: https://www.ctwm.org/dist/%{name}-%{version}.tar.gz
```
Comment (madonuko) [inline anda/desktops/ctwm/ctwm.spec:7]: So the annoying part is, we probably want to download sources directly from upstream (due to xz incident we should have learnt our lesson)

ctwm sources do not use git, they use bazaar (totally incompatible), commits are called revisions

supposedly we can track the version in update.rhai using this file https://bazaar.launchpad.net/~ctwm/ctwm/trunk/annotate/head:/VERSION
Downside is we need to regex the webpage to get the version
and then we need to grab the revision (710.1.2), save it in the spec using rpm.global(), then finally we have the tarball at https://bazaar.launchpad.net/~ctwm/ctwm/trunk/tarball/710.1.2

I might do this later

Comment (nothingneko) [reply anda/desktops/ctwm/ctwm.spec:7]: please do i cannot write rhai 

as for other changes i'll do my best and let you know if i need help :3

File: anda/desktops/ctwm/ctwm.spec (hunk 1 above)
Comment (nothingneko) [reply anda/desktops/ctwm/ctwm.spec:8]: cannot find it in rpmsphere sources ww

File: anda/desktops/ctwm/ctwm.spec [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+%build
+%cmake
+%cmake_build
+
+%install
+%cmake_install
+%{__install} -Dm644 %{SOURCE1} $RPM_BUILD_ROOT%{_datadir}/xsessions/%{name}.desktop
+
+%files
+%doc README.md CHANGES.md
```
Comment (madonuko) [inline anda/desktops/ctwm/ctwm.spec:28]:
```suggestion
%doc README.md CHANGES.md
%license COPYRIGHT
```

**Pull request closed without merging**